Guacamole/Assets/Scripts/MouseClick.cs

```diff
@@ -29,6 +31,12 @@ void OnMouseDown()
 
     void CreateMole()
     {
+        Respawn();
         mole.gameObject.SetActive(true);
     }
+
+    public void Respawn()
+    {
+        this.transform.position = spawnPoint.position;
+    }
```

Comment:
Is this "this" referring to the mole?

Reply:
In this case yes!
"This" refers to whichever object that the script is attached to.